lib/class-wp-theme-json-gutenberg.php

```diff
@@ -835,6 +837,10 @@ private function get_block_styles( $style_nodes, $setting_nodes ) {
 			$selector     = $metadata['selector'];
 			$declarations = self::compute_style_properties( $node );
 			$block_rules .= self::to_ruleset( $selector, $declarations );
+
+			if ( self::ROOT_BLOCK_SELECTOR === $selector ) {
+				$block_rules .= '.wp-site-blocks > * + * { margin-top: var( --wp-theme-block-gap ); margin-bottom: 0; }';
```

Comment:
Can this be the default value of `$block_rules`?

A theme may or may not add a `styles` key to its `theme.json`. What should happen with themes that don't have one? In such a case, if we still want to output this ruleset, it should be part of the default `$block_rules`. However, if we don't, it's fine as it is.

Reply:
I added a default value of `gap` in the default theme.json if that's what you mean? Is that different?

Reply:
I meant that, as it stands, this ruleset is not enqueued when the theme doesn't have any `style` because `style_nodes` will be empty (the code within the loop is not executed).

Reply:
But from my reading, it seems impossible for a theme to not have "style" since I added the default "style" to the default theme.json?

Reply:
Oh, I missed that, you're right. :+1: 

Anyway, it doesn't seem necessary to check for this in every cycle of the loop if we know we always want that ruleset in the stylesheet.

Reply:
@nosolosw Another question here. This style is injected properly in the template editor (post editor) but not in the site editor, do you know why and how to fix that?

Reply:
In addition to https://github.com/WordPress/gutenberg/pull/33812/files#diff-b03597cc3da199e5aa5a94950e8241135904853e7c3b82d758e42744878afae7R237 the property also needs to be declared in the client: `__EXPERIMENTAL_STYLE_PROPERTY` at `packages/blocks/src/api/constants.js` so the site editor can generate it as well.

Reply:
That's done already in the last commit, the CSS variable is generated but the style above is not included.

Reply:
Ah, I see. Adding the equivalent of this line https://github.com/WordPress/gutenberg/pull/33812/files#diff-b03597cc3da199e5aa5a94950e8241135904853e7c3b82d758e42744878afae7R842 in the client at https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/editor/global-styles-renderer.js#L308 should do it.